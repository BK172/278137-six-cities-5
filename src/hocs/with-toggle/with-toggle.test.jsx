import React, {Fragment} from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withToggle from "./with-toggle";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

const MockComponentWrapped = withToggle(MockComponent);

describe(`withToggle component test`, () => {
  it(`Should render withToggle correctly if isToggleActive true`, () => {
    const tree = renderer.create(
        <MockComponentWrapped
          isToggleActive={true}
          onToggleChange={()=>{}}
        >
          <Fragment />
        </MockComponentWrapped>
    )
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`Should render withToggle correctly if isToggleActive false`, () => {
    const tree = renderer.create(
        <MockComponentWrapped
          isToggleActive={false}
          onToggleChange={()=>{}}
        >
          <Fragment />
        </MockComponentWrapped>
    )
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
