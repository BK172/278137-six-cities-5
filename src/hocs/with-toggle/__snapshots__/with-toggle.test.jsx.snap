// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withToggle component test Should render withToggle correctly if isToggleActive false 1`] = `<div />`;

exports[`withToggle component test Should render withToggle correctly if isToggleActive true 1`] = `<div />`;
